Daybreak rework

KnockbackNemo opened the pull request:
# Quality Assurance Checklist
To make reviews more efficient, please make sure the board meets the following standards and check everything off once the board meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions.

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Minimum Prerequisites
- [x] The board passes ERC and DRC.
    - Note some ERC warnings are acceptable (**_CAUTION_** please be careful)
    - Note some DRC warnings are acceptable (less so than ERC: **_CAUTION_** please be careful).
- [ ] All traces are routed.
- [ ] Units are in metric.
- [ ] Gerbers are in a zip file in the main directory (after physical design is done)

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Schematic Level Requirements 
- [ ] Is proper noise resistance given to all peripheral devices (bypass caps and coils/ferrites)?  
- [ ] Is proper ESD protection given to all MCU input pins (zener diodes)? You could also write the input pin rating for the MCU on the schematic.  
- [ ] Is proper power protection given to peripheral devices (zener diodes)?  
- [ ] Are peripheral units used properly (reading datasheet)?  
- [ ] Are testing points added at useful places?  
- [ ] Is there proper short to GND protection at MCU outputs (inline resistors)?  
- [ ] Do ADC inputs have caps (appropriately sized to prevent RC delay)?  
- [ ] Are ADC inputs biased so there is room above expected value to determine if value is being overflowed?  
- [ ] Are LEDs located at useful places (comm, power, debugging, extra GPIO)? They should be placed in locations   
- [ ] Are parts chosen easy to collect?  
- [ ] Are parts chosen easy to solder?  
- [ ] Is there reverse polarity protection on inputs?  
- [ ] Do parts in the schematic match the actual part datasheet?

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## BOM Requirements
- [ ] **Bill Of Materials**: is there an Excel Sheet BOM and a Mouser Cart link?
- [ ] Are *all* the correct parts in the BOM?
- [ ] Is each part in the Mouser Cart in stock (or do we have the part in stock)?
- [ ] Does each part in the Mouser Cart have a margin for part quantity?
- [ ] Does every component in the schematic have a footprint, mouser link/datasheet, and reference number associated with it?

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Layout Level Requirements 
### 2D Spacing
- [ ] The components are spaced out at an optimal distance.
    - All components can be easily hand-soldered.
    - IPC-SM-782A Standard requires a minimum distance of 1.0mm from edge to edge.
- [ ] Components that are in parallel with each other are spaced out at an equal distance when possible.
- [ ] The components are aligned with each other when possible.
- [ ] Components are grouped based off of functionality.
    - E.g. all components for CAN should be grouped.
- [ ] Bypass capacitors are less than 1cm away from their respective IC's power pins.

### 3D Spacing
- [ ] Layout of components have been placed with mounting location and usage of the board in mind.
    - Are PCBs going to be stacked on above/below this board? Are tall components placed accordingly?
    - Are buttons and lights easily accessible and viewable?
    - When mounted into the car, are the heights of the components and connectors accounted for?
- [ ] I am not reading every word and I need to start this checklist over.
    - Please uncheck everything you have checked and start over.
- [ ] Location of connectors and wires going out of the board will prevent messy wiring.
    - Are all the wires that are going in the same direction placed on the same edge of the board?

### Components
- [ ] Do footprints match the schematic?
    - Make sure pins on the footprint correspond to pins in the schematic 
    - (e.g. counterclockwise pins on footprint and alternating pins on schematic is *bad*)
- [ ] Custom footprints have been double checked with the datasheet.
- [ ] Footprints are consistent across the PCB: *i.e. all 0805 for Resistors/Capacitors*
- [ ] Pin 1 of the footprint is labeled in some way. 
- [ ] Are LED's in easy to see places? 
- [ ] Are test points in easy to reach places?  
- [ ] Are critical paths of switching converters as small as possible?  

### Copper Layer 
- [ ] The trace widths and trace clearances are greater than JLCPCB's minimum requirements (https://jlcpcb.com/capabilities/Capabilities). 
    - [ ] Are signal traces are .15mm unless provided with reasoning?   
        - NOTE: One net can have multiple trace widths  
- [ ] The trace lengths are as short as possible. 
    - Can there be a more optimal route if you go to another layer? 
- [ ] Each trace's width is capable of handling the expected current flow. 
    - Use PCB width calculator to calculate trace width. 
- [ ] *No sharp corners. No trace angles equal to or less than 90 degrees. 
    - Orthogonal traces should have vias if necessary. 
- [ ] Are edges of board surrounded by clean ground on both layers with stitching vias?  
- [ ] Traces are in parallel with each other when possible (exceptions can be made to prevent signal coupling). 
    - E.g. traces connected between an IC and MCU can be placed in parallel with each other. 
- [ ] There are no random trace appendages. 
- [ ] Vias placed in copper pads are fully encompassed in the copper pads. 
- [ ] Through-hole components do not have extraneous vias. 
- [ ] Vias should be appropriately sized according to the connected net(s) and expected current capacity.

*Not really a problem for modern manufacturing techniques but good practice and important for high speed signal integrity.

### Silkscreen Layer
- [ ] Visible text sizes are greater than .8mm(L), .8mm(H), .15mm(W).
- [ ] All visible text is on the silkscreen layer.
- [ ] All reference labels of each component are not overlapping a copper pad or another component.
- [ ] All connector pins are labeled with a meaningful and helpful name.
- [ ] All LEDs are labeled with a meaningful and helpful name.
- [ ] Silkscreens are mostly facing one direction (or 2). 

### Edge Cut Layer
- [ ] The dimensions of the board and the mounting holes are nice values in metric i.e. no long decimals.
- [ ] The physical outline of the board is on the edge cut layer.
- [ ] Edge cuts are straight and parallel with opposite edge of the board.
- [ ] Mounting holes are aligned and support M3 screws. All mounting holes should have distances between them shown in comments layer.
- [ ] Corners are curved and contoured to mounting holes.

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## IMPORTANT NOTICE
- [ ] I am confident that this board will be safe to use in a safety critical environment.
- [ ] I have read every word on every bullet point and have only checked off a bullet point if I have read every word.
- [ ] ((( **OPTIONAL** ))) I had fun :^)

## Other Comments
_Write any comments about the board that would help the reviewers here._

Lakshay983 reviewed (changes requested):
Today I learned that KiCAD doesn't differentiate between net labels and hierarchical labels if they're named the same thing. Pressing ~ on the net shows that KiCAD thinks they are electrically connected: 
![image](https://github.com/user-attachments/assets/d6cce82c-0bfa-4643-ab8b-633ba59ee207) consider renaming the name of the nets to be different.

Some ERC stuff:
- the  "Foward" pin on the MCU isn't connected to anything (misspelling go brrrr)
- power labels are global so you don't need to have hierarchical labels. Either way, the relevant hierarchical labels aren't on the sheet.
![image](https://github.com/user-attachments/assets/cfaff912-4da2-4cd6-a4da-2066e52b2aa4)

The dashboard PCB has pin 2 of the dashboard connector as BPS hazard and y'all have it as regen.

I thought we were getting rid of the 555 timer in favor of doing it in software? Doing it in hardware is also fine, was just wondering.


Make sure you update all the footprints and ask for a review on that before you start layout.

You shouldn't need to use the "SI8261ABC-IS" gate drivers, you should be able to just use the EL3H7 (right @ppatra126 ?) The gate of the mosfet doesn't sink that much current.

Right_Ind and Left_Ind are at 12V logic and referenced to GNDPWR, so you can't plug it into the gate of your mosfet since the 5V of the timer is referenced to GND (thus breaking the isolation barrier).
![image](https://github.com/user-attachments/assets/dfbe3edf-0d09-495e-a24d-de3777a95760)

ppatra126 commented:
Would it be possible to break out another 4-pin connector for MotorCAN? In case we want to try adding any lighting/pedal boards onto this gen for testing that would allow us to daisy chain CAN through controls without the leader board being a terminating end. 

![image](https://github.com/user-attachments/assets/ae341464-c441-4dc8-9481-5663be5a351b)
We found that this configuration probably doesn't actually work since current can be conducted in the reverse direction through the body diode. The most recent contactor driver should have a proper configuration so I would copy their schematic.

KnockbackNemo commented:
Thank you both for the reviews!

Lakshay:
I renamed the nets to be different, chose to use global power labels, and renamed/added power flags until the ERC was happy. Regen is now Bps_Hazard, the timer uses an optoisolator instead of a mosfet (since it feeds directly into another optoisolator, I don't think current will be a problem?), and the SI8261ABC-IS gate drivers are now also optoisolator drivers.

We decided we like abstracting the blinky stuff into hardware, so we plan to keep the timer unless @diyarajon prefers software.

Prat:
I added a second CAN connector and changed the type to be the same one as that used by CarCAN. I also copied the power-in input circuit from the contactor board, though I'm not sure I understand the issue/solution. If you have time, would you mind explaining how the current goes backward through the diode?

Additional changes/concerns:

- I might be crazy, but it seemed like the optoisolators were reversed, so I flipped them all (and changed the resistor values to account for 12V logic). Could you check that they're being used correctly?
![image](https://github.com/user-attachments/assets/39d8c918-0e38-4bac-af22-f339bd5e5f54)
- Removed current-limiting resistor between the 555 timer and lights driver since the line feeds directly from one optoisolator to another, so the current should be within the allowable range already
- Not sure if I copied over and adapted the contactor board power distribution input circuit correctly
![image](https://github.com/user-attachments/assets/e85cdf25-df60-47a8-a486-730585f64add)
- MotorCAN no longer uses the shield lines. We didn't seem to be using them to begin with, but if you could check the MotorCAN circuit, that would be greatly appreciated.
![image](https://github.com/user-attachments/assets/bc0f68fa-4a82-40cf-b5ea-96d0c38b26f8)


Thank you very much!

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/1230abfd-c125-44a8-ac4a-c5937db279f4)
I would just remove the second set of potentiometers for brake/accel and only have one pot circuit per. Also if you're switching to a limit switch/pressure sensor or smth for brakes make sure that this circuit will continue to work as intended (if the pressure sensor is resistive or if it will work the same as a pot)

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/19f07fa7-a8e9-429f-a8b3-2013207dc939)
Do these just go to external LEDs? Is isolation required?

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/91a0fd81-b93f-4e21-8772-08e30041ba23)
![image](https://github.com/user-attachments/assets/1cdfbee1-8fc4-42ef-97bf-e7beefe6e734)
Do you intend to have 8 mounting holes or is this just something left over from copy/paste

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/462ba9bf-6f82-4084-875d-3095734883be)
Resistor value you use here will depend on the LED you use. We typically use APTD2012LCGCK (from [Standardized Mouser Components](https://utexas.sharepoint.com/:x:/r/sites/ENGR-LonghornRacing/_layouts/15/Doc.aspx?sourcedoc=%7BDDADD73C-3CE1-45DD-A622-EE3EAEDD15A2%7D&file=Standardized%20Mouser%20Components.xlsx&action=default&mobileredirect=true)). See https://www.sparkfun.com/tutorials/219 and verify this value will work.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/126e9c29-faa5-467e-bb9b-4af0f0e48bc6)
Where does this assumption come from? Where are the current limiting resistors going to be placed? I would make sure we have a plan for this, or if not we can just include the I-lim resistors on this board.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/f76febf6-7eb6-4556-8e78-f7320f2eb40c)
Would probably talk to Mika for DashboardPCB and spec out a LED (lowk it can just be one of them regular breadboard LEDs unless you really want something special)

IshDeshpa commented:
> Make sure you update all the footprints and ask for a review on that before you start layout.
I would definitely add a P/N (part number) and footprint to every component on here. Looks like the footprints are OK but double check that they're the ones you want.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/c04384d7-ff7f-4d79-b43d-eb55897f37e1)
I remember this being an issue; on the previous iteration the capacitor was labeled as 12pF but the part number links to https://www.mouser.com/ProductDetail/KEMET/C0805C479K1GACTU?qs=mC7y1gYrwOGjURrY82myhA%3D%3D which is 4.7pF (also an obsolete part)

Make sure the part number and the value align. Either it should be 4.7pF or it should be 12pF.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/7fea2007-6e57-4cc1-88e5-a746cdbe5fdf)
What does this mean? I'd probably make some of these comments a little more clear without context. If they're direct responses to things on the PR or responses to questions, rather than putting this stuff on the schematic IMO you could put it in the PR comments.

IshDeshpa reviewed (changes requested):
Good work guys!

Lakshay983 commented:
For motor CAN: 
- can you make the connector for Motor CAN the same one it is on the motor controller
- @ppatra126 when you say you want to add more boards to this bus, can you really do that? The psom assume 12V and CAN are referenced to the same GND. I think if we drove lights through a light peripheral we'd pop the motor CAN chip

I agree with Ishan on the points about notes. I think it's great y'all are writing reasoning on the schematic, but many of these without context are odd. Notes on design can be left on the schematic and questions/what changed between revisions can be on PR comments.

Minion board interface:
- can you change the name of the schematic
- The opto-isolators shouldn't be driving the actual LED, they should be driving the gate of a mosfet which then drives an LED. The gate of a mosfet takes very low current to turn activate
- Inputs wise 6mA should be fine (double check with the STM32F4 datasheet if you'd really like to make sure). Either way you can still make the resistors smaller to drive more current

I just realized, the linked datasheet for the EL3H7 is for a different part number: https://www.vishay.com/docs/83740/sfh617a.pdf

![image](https://github.com/user-attachments/assets/534a2824-3eae-4ba9-8182-d0fd87330106)
where do Right_Ind and Left_Ind go in the Lights schematic? 

![image](https://github.com/user-attachments/assets/f90d9c48-e7e7-4ff4-b336-b9aa5096df8d)
Shouldn't Timer_CLK be driving the LED?

Can you use the IRLML0030  for your mosfets <- new standardized mosfet 

![image](https://github.com/user-attachments/assets/9de52f59-c3de-4e29-86c7-edbbe24ecf17)
Power labels are global so you don't need hierarchical labels

Lakshay983 commented:
Just talked to @ppatra126, you can take out the extra motor CAN connector.

Just have it be a 6 pin one the same pinout as the motor controller (with Controls on the terminating end).

KnockbackNemo commented:
> ![image](https://private-user-images.githubusercontent.com/22435311/382493997-7fea2007-6e57-4cc1-88e5-a746cdbe5fdf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NzQxODEsIm5iZiI6MTczMDU3Mzg4MSwicGF0aCI6Ii8yMjQzNTMxMS8zODI0OTM5OTctN2ZlYTIwMDctNmU1Ny00Y2MxLTg4ZTUtYTc0NmNkYmU1ZmRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAyVDE4NTgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNjczNzY2NzZjMWQ1NzQyMWM4YzI5YmYzMmJmOWJkNzE4ODVhMzM1YWJlZjIwZWI3OWZlOTQzMmNkMjYyMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rG2U3gg6s53gpXB-_yAnx4WkY89s_SDOx-KsSl5Xd-w) What does this mean? I'd probably make some of these comments a little more clear without context. If they're direct responses to things on the PR or responses to questions, rather than putting this stuff on the schematic IMO you could put it in the PR comments.

We were leaving notes for ourselves on the schematic as we went. We'll rewrite or remove them for the actual board now that they've been clarified

KnockbackNemo commented:
> where do Right_Ind and Left_Ind go in the Lights schematic?
> Shouldn't Timer_CLK be driving the LED?

The way we have it, Timer_CLK is a software version of the 555 timer that blinks constantly, and Right_Ind and Left_Ind are sent from the dashboard and determine whether or not that signal passes through to the right and left lights drivers respectively. We could change it so the microcontroller does logic with switch inputs and then controls the lights using a blink-right/blink-left signal, but then we'd need to have optoisolators/GPIO inputs for the left switch position, right switch position, and hazard switch

KnockbackNemo commented:
> ![image](https://private-user-images.githubusercontent.com/22435311/382493248-126e9c29-faa5-467e-bb9b-4af0f0e48bc6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMzUxNjgsIm5iZiI6MTczMTAzNDg2OCwicGF0aCI6Ii8yMjQzNTMxMS8zODI0OTMyNDgtMTI2ZTljMjktZmFhNS00NjdlLWJiOWItNGFmMGYwZTQ4YmM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDAzMDEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyY2JkZTUwNDQzMDRiZTQ0N2JiNDE1ZTQ2Y2U5YjFjOWNhNTUxNGU5NzY1MzQ4YWZhMjgxYzVmNDMyMDA1YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y5gmfzsGWucgUDyXSsKdZNSkl0Qup7LhJ6Gncs5GIfI) Where does this assumption come from? Where are the current limiting resistors going to be placed? I would make sure we have a plan for this, or if not we can just include the I-lim resistors on this board.

This was a legacy comment from the previous iteration, but if we're supplying the correct voltage to the external lights, will they not have the correct resistance to limit current and avoid blowing up?

KnockbackNemo commented:
Is there a 

> ![image](https://private-user-images.githubusercontent.com/22435311/382492910-462ba9bf-6f82-4084-875d-3095734883be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMzUxNjgsIm5iZiI6MTczMTAzNDg2OCwicGF0aCI6Ii8yMjQzNTMxMS8zODI0OTI5MTAtNDYyYmE5YmYtNmY4Mi00MDg0LTg3NWQtMzA5NTczNDg4M2JlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDAzMDEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmMTU4NzUxMGMzNjRjNWViOTllZTJjODBjY2RhODU3ZWUxZjM1YTRkM2E3NTlmMzZhN2M1ZTMzNDM2MDQ4ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZTYojN7bb19h9rRLGL8Kd0LHCkT8T5LvpFtVdRcigJg) Resistor value you use here will depend on the LED you use. We typically use APTD2012LCGCK (from [Standardized Mouser Components](https://utexas.sharepoint.com/:x:/r/sites/ENGR-LonghornRacing/_layouts/15/Doc.aspx?sourcedoc=%7BDDADD73C-3CE1-45DD-A622-EE3EAEDD15A2%7D&file=Standardized%20Mouser%20Components.xlsx&action=default&mobileredirect=true)). See https://www.sparkfun.com/tutorials/219 and verify this value will work.

Is this where we should go to choose components for commonly-used elements? I might just have missed it, but a lot of the parts listed here don't seem to match the ones we currently use? (IRLML0030, EL3H7)

KnockbackNemo commented:
> Make sure the part number and the value align. Either it should be 4.7pF or it should be 12pF.

I'm guessing it was probably supposed to be 12pF, but this is something we took from the previous iteration as well. Is there some convention we typically follow for choosing capacitor values?

Lakshay983 commented:
> > ![image](https://private-user-images.githubusercontent.com/22435311/382493248-126e9c29-faa5-467e-bb9b-4af0f0e48bc6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMzUxNjgsIm5iZiI6MTczMTAzNDg2OCwicGF0aCI6Ii8yMjQzNTMxMS8zODI0OTMyNDgtMTI2ZTljMjktZmFhNS00NjdlLWJiOWItNGFmMGYwZTQ4YmM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDAzMDEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIyY2JkZTUwNDQzMDRiZTQ0N2JiNDE1ZTQ2Y2U5YjFjOWNhNTUxNGU5NzY1MzQ4YWZhMjgxYzVmNDMyMDA1YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y5gmfzsGWucgUDyXSsKdZNSkl0Qup7LhJ6Gncs5GIfI) Where does this assumption come from? Where are the current limiting resistors going to be placed? I would make sure we have a plan for this, or if not we can just include the I-lim resistors on this board.
> 
> This was a legacy comment from the previous iteration, but if we're supplying the correct voltage to the external lights, will they not have the correct resistance to limit current and avoid blowing up?

It's honestly fine, LED strips normally have current limiting. If you really want to make sure you can add a 0 ohm resistor in series but I think it's fine

Lakshay983 commented:
> > Resistor value you use here will depend on the LED you use. We typically use APTD2012LCGCK (from [Standardized Mouser Components](https://utexas.sharepoint.com/:x:/r/sites/ENGR-LonghornRacing/_layouts/15/Doc.aspx?sourcedoc=%7BDDADD73C-3CE1-45DD-A622-EE3EAEDD15A2%7D&file=Standardized%20Mouser%20Components.xlsx&action=default&mobileredirect=true)). See https://www.sparkfun.com/tutorials/219 and verify this value will work.
> 
> Is this where we should go to choose components for commonly-used elements? I might just have missed it, but a lot of the parts listed here don't seem to match the ones we currently use? (IRLML0030, EL3H7)

This document is kinda out of date. Just try to use it for passive components like caps or resistors and ignore most other things. Stuff like the IRLML0030 is a new development and not on there. Before you're done with BOM you can just upload it to mouser to see if any of the components are bad (don't include molex connectors on the mouser BOM).

You can do it at the end or now if you'd like, it comes down to personal preference. I've seen people prefer to do BOMs during schematic or all at the end.

Lakshay983 commented:
> > Make sure the part number and the value align. Either it should be 4.7pF or it should be 12pF.
> 
> I'm guessing it was probably supposed to be 12pF, but this is something we took from the previous iteration as well. Is there some convention we typically follow for choosing capacitor values?

There's some wacky frequency math you can do to figure it out on the STM32 datasheet. I'd just probe those capacitors on the current Controls Leader that works and use that value. If it worked for the last revision then it'll work again for this revision.

Lakshay983 commented:
> > where do Right_Ind and Left_Ind go in the Lights schematic?
> > Shouldn't Timer_CLK be driving the LED?
> 
> The way we have it, Timer_CLK is a software version of the 555 timer that blinks constantly, and Right_Ind and Left_Ind are sent from the dashboard and determine whether or not that signal passes through to the right and left lights drivers respectively. We could change it so the microcontroller does logic with switch inputs and then controls the lights using a blink-right/blink-left signal, but then we'd need to have optoisolators/GPIO inputs for the left switch position, right switch position, and hazard switch

Ah I see, that's clever. Nice!

Also, why no current limiting resistors on R_Blink and the L_Blink optoisolators?

Lakshay983 commented:
> > > Make sure the part number and the value align. Either it should be 4.7pF or it should be 12pF.
> > 
> > 
> > I'm guessing it was probably supposed to be 12pF, but this is something we took from the previous iteration as well. Is there some convention we typically follow for choosing capacitor values?
> 
> There's some wacky frequency math you can do to figure it out on the STM32 datasheet. I'd just probe those capacitors on the current Controls Leader that works and use that value. If it worked for the last revision then it'll work again for this revision.

You can also check the JLCPCB bom branch I made to see what value I used for the caps, it supposedly turned on 🤷 . If you have the Controls Leader I'd just try probing the capacitance

krazykrayon commented:
All the above comments (I think) have been addressed.

KnockbackNemo commented:
@FrankLiTX @ParthivS20 Our connectors currently all have the clips facing out (like the last Controls leaderboard). Should we flip them to face the inside instead?

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/05cdcab3-d879-4fc6-aad9-2dfde31b814c)
Make sure these holes don't go off the edge cut border

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/cf34c318-074c-4adb-8777-9ea4fa4562f0)
is this a cloud or a tree

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/d7623bec-f2b5-47d9-a4a2-88b624dfe7fe)
This seems a bit unwieldy. Could you put the fuse on the back and then move some traces so that this trace is a bit more direct? it just sucks this is the one thing that's like on the other end of the board

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/44577437-54a2-4740-8f04-bf1b7b0967d4)
Can you explain the brake pot AND brake switch decision? why both, not sure i remember.

IshDeshpa commented:
Also for the 12V traces you do have double check to make sure that they can handle the required amount of current.

IshDeshpa reviewed (commented):
That's probably all of my comments, it's not much but that's just because I'm not used to reviewing PCBs. If a @Lakshay983 or a @Champers5000 or a @FrankLiTX could review this for actual design errors it may be better.

Lakshay983 commented:
> ![image](https://private-user-images.githubusercontent.com/22435311/401326887-44577437-54a2-4740-8f04-bf1b7b0967d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzODQ5NjEsIm5iZiI6MTczNjM4NDY2MSwicGF0aCI6Ii8yMjQzNTMxMS80MDEzMjY4ODctNDQ1Nzc0MzctNTRhMi00NzQwLThmMDQtYmYxYjdiMDk2N2Q0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDAxMDQyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmZDJjODE5ZDE1ZTUzNjEyMDFjMzY3OTgyNzM1OWVkZTJkY2I1MzJiMzkyNzc0MDU4NjEwZGY4ODM3NWJjNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UyWNriNiWxRSu1pyuiIVY-zHxkQLdAbt08BvlK-eyKU) Can you explain the brake pot AND brake switch decision? why both, not sure i remember.

Iirc it was because Ergo didn't know 100% which option they were doing for Daybreak and it was easier to support both on the Controls side.

Lakshay983 commented:
Can you make J1 (the extra ADC connector) a molex connector since the pin headers aren't great for vibration

C41 is a bypass cap for the usb-uart chip and it's pretty far away, can you make it closer to the 3v3 pin of that chip

Make sure you update your board constraints: https://github.com/lhr-solar/UTSVT-KiCadLibraries?tab=readme-ov-file#standard-constraints-for-laying-out-board since I'm running DRC and it's giving me a ton of nets bridged since your clearance requirement is higher than needed

Can you fill some of the areas that aren't filled. It's mostly for ✨ aesthetic ✨ 
<img width="361" alt="Screenshot 2025-01-08 at 10 09 27 PM" src="https://github.com/user-attachments/assets/2059615e-4922-4c05-b93b-5860f8e7a7c3" />
<img width="347" alt="Screenshot 2025-01-08 at 10 10 01 PM" src="https://github.com/user-attachments/assets/bff83377-f3e6-4f81-be4b-0d27e1494be4" />
The solution usually being to group these closer traces to be in busses

I second Ishan on the 12V trace being a bit odd. I also don't like routing all the current for the lights technically next to the crystal oscillator. Maybe try routing it around the border of the board to get to motor connector. There's also the nuclear option of switching the carcan and motorcan stuff is so you don't need to route the 12V trace as much.

The brake pot trace is longer than it needs to be. I think you can move C8 slightly closer to the oscillator and you can fit the trace inside that bus of red traces to go directly into the chip
<img width="756" alt="Screenshot 2025-01-08 at 10 15 16 PM" src="https://github.com/user-attachments/assets/03c36164-e26e-4514-ad10-c206869bb4de" />

For the 3v3 traces, you can simplify a lil bit since there's some routing on the top and bottom layer going in different directions.
<img width="813" alt="Screenshot 2025-01-08 at 10 18 54 PM" src="https://github.com/user-attachments/assets/078283dc-1607-47f8-8a8a-6d20f24dcf7e" />
You could just via from that red tree branch up to the top pin. 
<img width="556" alt="Screenshot 2025-01-08 at 10 21 31 PM" src="https://github.com/user-attachments/assets/cbd86f76-7336-4c47-a6c2-b8610a677a9e" />

you can get rid of this GND trace and via off of pin 47 and C11 by moving the SWDIO via out a little bit so you can form a thermal relief. 
<img width="396" alt="Screenshot 2025-01-08 at 10 23 17 PM" src="https://github.com/user-attachments/assets/85f5918e-8fd1-40e5-a179-80669558f2e9" />

For pin 63 and 12 instead of a via and long trace you could just do a small trace to a via per pin. I could spin some wacky signal integrity reason like you want both pins to have equal traces to the GND plane but it's mostly for aesthetic and reduction of traces.

CAN routing can be improved a little bit so the signal traces aren't branching in a bunch of directions. 
<img width="685" alt="Screenshot 2025-01-08 at 10 36 01 PM" src="https://github.com/user-attachments/assets/a819dce8-260e-4148-833a-fb29a82764c5" />
Flip R34, move U18 up a little bit, swap R35 and JP2 so now the can traces follow more of a path and the area can fill
<img width="630" alt="Screenshot 2025-01-08 at 10 43 25 PM" src="https://github.com/user-attachments/assets/8dcbf2eb-e48a-46e0-87d2-4efbce4427a6" />
With that you can also route the GNDISO trace down the middle of the chip and into the pins instead of routing it around the chip and through the resistors.
<img width="824" alt="Screenshot 2025-01-08 at 10 54 52 PM" src="https://github.com/user-attachments/assets/5778594f-52af-4724-8d50-26d7e71fb8cb" />
This was my idea for routing the CAN signals so there's some less long branching but also more vias. Up to y'all on what you want to do tho.
Most of this stuff applies for the motor can chip as well.

Why route the break switch through the reset button? Could be routed under the reset button mayhaps.

Motor should be GNDPWR not GND
<img width="735" alt="Screenshot 2025-01-08 at 10 37 14 PM" src="https://github.com/user-attachments/assets/9affa391-1c13-471e-b4c3-fc117f2ca4b5" />

Lakshay983 commented:
Be careful about PC13-PC15. The datasheet specifies some limitations for those pins
<img width="1071" alt="Screenshot 2025-01-08 at 11 30 51 PM" src="https://github.com/user-attachments/assets/a4244cce-7284-4fbc-8654-fd8aabdad63b" />

Lakshay983 commented:
Why do you have the area with no fill? You can extend the GND fill to fill that space. Also, i'd suggest making the GND plane a square that fills the entire board, and then assigning it a lower priority than your 12V and GNDPWR planes so the 12V plane will always fill where it needs to and everywhere else it'll fill GND. Just so you don't need to draw wonky shapes if you ever need to make stuff around.
<img width="330" alt="Screenshot 2025-01-09 at 1 07 44 PM" src="https://github.com/user-attachments/assets/277f46c9-a899-4fbc-b828-860b2c1fca17" />

Lakshay983 commented:
Also i don't think you need the 2 CAN testpoints if you have those DB9 outputs, it's also fine to keep em.

Lakshay983 commented:
<img width="595" alt="Screenshot 2025-01-09 at 1 25 12 PM" src="https://github.com/user-attachments/assets/a06faf48-b25a-41d8-b090-92cb165f23db" />
This breaklight trace is pretty small considering that it's going to power multiple lights and it's all returning through that small trace. You have a big plane to sink all that current for GNDPWR but you're limiting yourself with that trace going into the fet.

Cam0Cow commented:
![image](https://github.com/user-attachments/assets/f76ab391-52d0-4057-8f5d-5debaeb027dd)
This seems overkill for what I assume is an on-board LED, or is it routed elsewhere to something beefier?

![image](https://github.com/user-attachments/assets/000d705e-f98f-425d-bf90-0e18907049e5)
Are these resistor values accurate? It looks just like the old circuit, but you're wasting a decent amount of the range of the ADC which doesn't seem ideal. A weaker pull-up might make sense here.

![image](https://github.com/user-attachments/assets/86d523cc-6e1a-4862-b2ce-1a7017ebc4c2)
Not gonna nitpick the layout since it seems Lakshay has that covered, but it seems like there's still some missing connections, like this 12V pin. Are all the other new boards two layer only as well, or just this one? It doesn't seem to necessarily need it, just curious.

KnockbackNemo commented:
> ![image](https://private-user-images.githubusercontent.com/22435311/401324946-cf34c318-074c-4adb-8777-9ea4fa4562f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2MTk3MDQsIm5iZiI6MTczNjYxOTQwNCwicGF0aCI6Ii8yMjQzNTMxMS80MDEzMjQ5NDYtY2YzNGMzMTgtMDc0Yy00YWRiLTg3NzctOWVhNGZhNDU2MmYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTExVDE4MTY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxY2NlZjg1NTEzYzAwNTdjZTZiMzA0NDM5ZmY4ZmJlMWU1OTE3ZThjMzFmZGE3NGJiZGJmY2E0YmFiZGVlNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hlibUh5qknYGfwKsU7fnKDMuK03OhD2ObeBMdTIoosI) is this a cloud or a tree



![IMG_8470 (2)](https://github.com/user-attachments/assets/07934eba-424a-479c-9c6f-c0e685778ee0)

It's a masterpiece.

Or a plea for help.

I'm not sure which.

KnockbackNemo commented:
@Lakshay983 Thank you for the suggestions! I'm still getting some "nets bridged" errors for certain components, even with the updated clearance requirements. Should I lower the reqs until they go away or exclude the errors?

KnockbackNemo commented:
> ![image](https://private-user-images.githubusercontent.com/11470175/401789956-f76ab391-52d0-4057-8f5d-5debaeb027dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2MTk3MDQsIm5iZiI6MTczNjYxOTQwNCwicGF0aCI6Ii8xMTQ3MDE3NS80MDE3ODk5NTYtZjc2YWIzOTEtNTJkMC00MDU3LThmNWQtNWRlYmFlYjAyN2RkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTExVDE4MTY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzMjQxNzliZDk1MmViN2Y3ODRlZTNiOTY5ZDc2ZGY5OTM5NmMyYmRjNzZjYjM4NWYxN2ZiNTg5OTljNTU4NjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-lWjJ8f0y_i_N7KjS5JMHezpYbTFatbyIkiPcVY4OCQ) This seems overkill for what I assume is an on-board LED, or is it routed elsewhere to something beefier?
> 
> ![image](https://private-user-images.githubusercontent.com/11470175/401790554-000d705e-f98f-425d-bf90-0e18907049e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2MTk3MDQsIm5iZiI6MTczNjYxOTQwNCwicGF0aCI6Ii8xMTQ3MDE3NS80MDE3OTA1NTQtMDAwZDcwNWUtZjk4Zi00MjVkLWJmOTAtMGUxODkwNzA0OWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTExVDE4MTY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZlNmFkYjZjMTJlNTczZTM1MDVjMDRlN2E0ZDdlYmU4NmIwZWM2N2Q4ODllODAyY2EwYWRjMTVkYWEzMDJhZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GMtEHcKWRfF0Cgd8zMozYVsi1NHfQmoB8g5ja6l44yM) Are these resistor values accurate? It looks just like the old circuit, but you're wasting a decent amount of the range of the ADC which doesn't seem ideal. A weaker pull-up might make sense here.
> 
> ![image](https://private-user-images.githubusercontent.com/11470175/401792295-86d523cc-6e1a-4862-b2ce-1a7017ebc4c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY2MTk3MDQsIm5iZiI6MTczNjYxOTQwNCwicGF0aCI6Ii8xMTQ3MDE3NS80MDE3OTIyOTUtODZkNTIzY2MtNmUxYS00ODYyLWIyY2UtMWE3MDE3ZWJjNGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTExVDE4MTY0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxOTZmMmI4MjMzMmM2MTljYzIyYmU5MTU4YjRiNWUwNDI0ODhkN2Y3Y2QyNGYwNTQ3MGMxNTlmZTNmNTU2MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OJfBzEqDLsBzpd9MV9kSE5xkTeCWyWGA0mzuuUQLQ10) Not gonna nitpick the layout since it seems Lakshay has that covered, but it seems like there's still some missing connections, like this 12V pin. Are all the other new boards two layer only as well, or just this one? It doesn't seem to necessarily need it, just curious.

Roie returns! Thank you for the review.

1) Dbg led goes to the dashboard from the leaderboard, and everything on the dashboard is 12V; that's the reason for the whole setup.

2) That's a good point. The circuit was copied from the old board, but I see what you mean about the range being wasted. Do you have recommendations on choosing new values or know how the old ones were chosen? If I'm looking at the circuit right, I would think the lower both resistance values compared to the potentiometer, the larger the range of the ADC, so I'm assuming there are other reasons why we chose values as large as we did?

3) Whoops, thanks for catching that! I'll be more careful about reading the DRC errors. I'm not certain about the layers in the new boards, but @Lakshay983 would know better

KnockbackNemo commented:
Ok! Also actually never mind looking into the "net bridged" stuff- I thought they were errors, but I think all the ones left aren't really errors since it's for the surrounding aperture thing and not the nets themselves.

Lakshay983 commented:
Also can you make that one 45 degree cap not 45 degrees, it's weird that it's the only thing that's 45

KnockbackNemo commented:
![image](https://github.com/user-attachments/assets/6405a0ad-3f39-4b33-a7be-f9d5cb68f915)
I changed the accel pot resistors to 100k pulldown and 4.9k current limiting, and I left the brake pot circuit as it is. From my scribbled attempts, I think this still works to keep current down and increase the range, but please let me know if I messed something up.

Lakshay983 commented:
Avoid the via in pad for a pad this small (will make electrical contact when soldering hard
![image](https://github.com/user-attachments/assets/c1f9803f-5607-4c57-8fbf-f5746a772345)

I also would not recommend a via bridging two pads:
![image](https://github.com/user-attachments/assets/65ab55fd-a1ae-4a68-82d5-38d4316acb55)

Lakshay983 commented:
![image](https://github.com/user-attachments/assets/062c66b0-e602-446e-8616-b5137d02407d)
There's an unconnected item

Lakshay983 commented:
Can you fix wtv setting is giving you all those bridging nets violation?

Can you move Array_PBC_Sense up a little bit so there's a clean fill in the MCU gnd fill

Lakshay983 commented:
Other than that lgtm

KnockbackNemo commented:
I got rid of the errors by allowing bridged solder mask apertures between pads, so the footprints no longer match the library, but KiCAD doesn't complain anymore. 

I wasn't sure where Array_PBC_Sense was causing issues, so I just looked over the fills and tried to see if there were any areas I could improve.

I'll work on the BOM and other images next. Thanks so much for all your help!

Lakshay983 reviewed (approved):
![IMG_6908](https://github.com/user-attachments/assets/3569fe2b-dbd6-4ae8-a201-6dce1790f4cb)
